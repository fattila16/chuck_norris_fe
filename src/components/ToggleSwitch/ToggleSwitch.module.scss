@import 'src/assets/scss/color-palette';
@import 'src/assets/scss/vars';

.input[type=checkbox]{
	height: 0;
	width: 0;
	visibility: hidden;
}

.label {
    cursor: pointer;
    font-size: 18px;
    color: white;
    text-indent: 75px;
    text-align: center;
	width: 70px;
	height: 30px;
	background: map-get($active-theme ,'grey' );;
	display: block;
	border-radius: 70px;
    position: relative;
    top: -10px;
}

.label:after {
	content: '';
	position: absolute;
	top: 0px;
	left: 0px;
	width: 30px;
	height: 30px;
	background: map-get($active-theme , silk );
	border-radius: 30px;
	transition: 0.3s;
}

.input:checked + label {
	background: map-get($active-theme ,fawn );
}

.input:checked + label:after {
	left: calc(100%);
	transform: translateX(-100%);
}

.label:active:after {
	width: 130px;
}

@media only screen and (max-width: $mobile-breakpoint) {
	.label {
		width: 50px;
		height: 20px;
		position: relative;
	}
	.label:after {
		width: 20px;
		height: 20px;
	}
}