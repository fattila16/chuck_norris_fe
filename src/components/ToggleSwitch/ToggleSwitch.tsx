import React, { ChangeEvent } from 'react';
import css from './ToggleSwitch.module.scss';

interface ToggleSwitchProps {
    onChange?: (event: ChangeEvent<HTMLInputElement>) => void;
    checked?: boolean;
}
const ToggleSwitch: React.FC<ToggleSwitchProps> = (props: ToggleSwitchProps) => {
    const { onChange, checked } = props;
    return (
        <>
            <input className={css.input} type="checkbox" id="switch" onChange={onChange} checked={checked} />
            <label className={css.label} htmlFor="switch">
                <span>{checked ? 'RTL' : 'LTR'}</span>
            </label>
        </>
    );
};

export default ToggleSwitch;
