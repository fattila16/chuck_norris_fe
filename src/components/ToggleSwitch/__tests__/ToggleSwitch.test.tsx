import React from 'react';
import { shallow } from 'enzyme';
import ToggleSwitch from '../ToggleSwitch';

describe('[ToggleSwitch]', () => {
    it('Should render ToggleSwitch component without error', () => {
        const component = shallow(<ToggleSwitch />);
        expect(component.getElements()).toMatchSnapshot();
    });
});
