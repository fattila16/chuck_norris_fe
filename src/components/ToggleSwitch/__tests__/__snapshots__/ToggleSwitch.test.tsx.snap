// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ToggleSwitch] Should render ToggleSwitch component without error 1`] = `
Array [
  <React.Fragment>
    <input
      className="input"
      id="switch"
      type="checkbox"
    />
    <label
      className="label"
      htmlFor="switch"
    >
      <span>
        LTR
      </span>
    </label>
  </React.Fragment>,
]
`;
